src/ripple/app/tx/impl/BookTip.cpp

```diff
@@ -47,23 +48,24 @@ BookTip::step ()
         // See if there's an entry at or worse than current quality. Notice
         // that the quality is encoded only in the index of the first page
         // of a directory.
-        auto const first_page (view().getNextLedgerIndex (m_book, m_end));
+        auto const first_page =
+            view_.succ (m_book, m_end);
 
-        if (first_page.isZero())
+        if (! first_page)
             return false;
 
-        unsigned int di (0);
-        SLE::pointer dir;
+        unsigned int di = 0;
+        std::shared_ptr<SLE> dir;
 
-        if (view().dirFirst (first_page, dir, di, m_index))
+        if (dirFirst (view_, *first_page, dir, di, m_index))
```

Comment:
It looks like `dir` could be `std::shared_ptr<SLE const>`, then the call to `dirfirst` could be `cdirfirst` instead. This would be the last reference to `dirFirst`. This change builds.